### PR TITLE
Implement linearhash new

### PR DESCRIPTION
**Pull request by spencermacbeth**

### CODE REVIEW

File: src/dictionary/linear_hash/linear_hash_types.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+/* Created by Spencer MacBeth on 2017-02-12. */
+/*  */
+#include <stdio.h>
+#include "../../key_value/kv_system.h"
+#include "../dictionary.h"
+#include "../../file/SD_stdio_c_iface.h"
+
+typedef ion_byte_t *linear_hash_record_status_t;
+
+#define linear_hash_record_status_empty 0;
```
Comment (Stickerpants) [inline src/dictionary/linear_hash/linear_hash_types.h]: Constants should be ALL_CAPS_UNDERSCORES. No semicolon after defines.

File: src/dictionary/linear_hash/linear_hash_handler.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	return linear_hash_delete(key, (linear_hash_table_t *) dictionary->instance);
+}
+
+ion_err_t
+linear_hash_delete_dictionary(
+	ion_dictionary_t *dictionary
+) {
+	ion_err_t result = linear_hash_destroy((linear_hash_table_t *) dictionary->instance);
+
+	free(dictionary->instance);
```
Comment (Stickerpants) [inline src/dictionary/linear_hash/linear_hash_handler.c]: Mis-read, not actually a bug - no change to be made here

File: src/dictionary/linear_hash/linear_hash.c [hunk 3]
```diff
@@ -0,0 +1,1364 @@
...
+
+			if (err_ok != status.error) {
+				return status;
+			}
+
+			/* Set the location of the anchor record on the new overflow bucket and update the record_loc for storing
+			 * the new record to be this location */
+			ion_fpos_t overflow_anchor_record_loc = get_bucket_records_location(*overflow_location);
+
+			status.error = linear_hash_get_bucket(*overflow_location, &bucket, linear_hash);
```
Comment (spencermacbeth) [inline src/dictionary/linear_hash/linear_hash.c]: no, good catch! the update below was not necessary either as it was handled below.

File: src/dictionary/linear_hash/linear_hash_handler.h [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	ion_dictionary_t	*dictionary,
+	ion_key_t			key,
+	ion_value_t			value
+);
+
+#if defined(__cplusplus)
+}
+#endif
+
+#endif /* SKIP_LIST_HANDLER_H_ */
```
Comment (Stickerpants) [inline src/dictionary/linear_hash/linear_hash_handler.h]: Still needs fix

File: src/dictionary/linear_hash/linear_hash_handler.h [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+	ion_key_t			key,
+	ion_value_t			value
+);
+
+/**
+@brief	  Creates an instance of a dictionary.
+
+@details	Creates an instance of a dictionary given a @p key_size and
+			@p value_size, in bytes as well as the @p dictionary_size, which
+			is the maximum number of levels in the skiplist. By nature of the
```
Comment (Stickerpants) [inline src/dictionary/linear_hash/linear_hash_handler.h]: still needs fix

**Comment by danaack** [comment]
I agree that it's ready for development without cursor queries 👍 Looks good!

**Comment by Stickerpants** [comment]
Branch is ready for merge, just needs to be updated with the latest development, and then @spencermacbeth has the OK to complete the merge.

**Pull request merged**